lib/cb/utils/api.rb

```diff
@@ -57,10 +57,14 @@ def cb_delete(path, options = {}, &block)
       def execute_http_request(http_method, uri, path, options = {}, &block)
```

Comment:
I'm not opposed to doing that, but that's larger than this PR. What I'd imagine for breaking this apart would be breaking `exeucute_http_request` to `wrapped_http_request` and `execute_http_request`.
`wrapped_http_request`: Does the timing setup portion and calls out to `execute_http_request`.
`execute_http_request`: Does the request itself and the response validation call.

So I say future optimization there.